src/log.cpp

```diff
@@ -274,8 +275,39 @@ void Logger::log(LogLevel lev, const std::string &text)
 	const std::string label = getLevelLabel(lev);
 	const std::string timestamp = getTimestamp();
 	std::ostringstream os(std::ios_base::binary);
+
+	static const std::string use_logcolour = g_settings->get("log_colour");
+
+	bool coloured = use_logcolour == "tty" ? isatty(STDOUT_FILENO) : use_logcolour == "yes";
```

Comment:
Doesn't seem to be possible: http://stackoverflow.com/questions/5156675/#answer-5157038

What would (probably) be possible is for the class to remember whether it was redirected to a file.